### PR TITLE
Fix the issue with folder names starting with dots

### PR DESCRIPTION
**Pull request by otanim**
If you try to use `instaloader` to retrieve highlights that are grouped by naming it `.` or `..` you'll be having a problem with your OS. Besides, files starting with dots in Linux OS will be considered hidden folders, and you'll have some trouble seeing/moving them unless you use extra commands.

This commit uses the existing folder naming sanitization function and extends its capabilities for the above-mentioned case.

Basically, it will replace folders that start with a dot (".") into another similar one ("․")․

### CODE REVIEW

**Comment by Thammus** [comment]
The purpose of the `sanitize_path()` function that your pull request touches is merely to alleviate problems caused by characters that are invalid as part of filenames under Windows. As far as I know folders and filenames starting with a dot do not lead to errors on common OS. However, the described problem that users might get confused by hidden folders is a valid case.
Instaloader has a lot known Issues related to pathnames such as #185, #575, #710 and #859. Like described in https://github.com/instaloader/instaloader/pull/854#issuecomment-727220601, it would be nice to have a flag `--sanitizepaths` to do a proper path sanitation without having to worry about issues with backwards incompatibility.
Nevertheless, I think that substituting leading periods in pathnames by a similar looking character is a change that has more advantages than it might cause problems.

**Comment by otanim** [comment]
You're welcome.
